### DISCUSSION TITLE
How much resources is necessary to run a Kustomize?

### DISCUSSION THREAD

**Discussion by joaocarvalhowd** [argoproj/argo-cd#6231] [Q&A]
I have set some limits for CPU and Memory. And after applied that limit, the ArgoCD with Kustomize, stop working. 

<img width="563" alt="image" src="https://user-images.githubusercontent.com/7674544/118164850-f2ffa180-b3f9-11eb-90a0-0fce2a5bd8f4.png">

![image](https://user-images.githubusercontent.com/7674544/118165033-322df280-b3fa-11eb-9b06-6a51b4af44a0.png)

Does someone know how much Memory and CPU is necessary?

**Comment by jannfis** [comment]
Hi @joaocarvalhowd, all tooling is executed in the `argocd-repo-server`  workload, not in the `argocd-application-controller`.

Also, manifest generation errors get cached for some time (see the `(cached)`  indicator in the error message). You may force to clear the cache by requesting a hard refresh of the application, instead of a normal refresh.

**Reply by jessesuen** [reply]
To add to what @jannfis said, there is also a tunable to the repo server `--parallelismlimit` which limits concurrent commands that can be run (e.g. `kustomize build`). The default is 0 (no limit). You may need to tune this value if you find that your repo-server access pattern is bursty and needs to run `kustomize build` concurrently some times. Note that parallelismlimit acts like a semaphore, so setting it too low will cause delays in manifest generation if there a backlog of manifest generate requests. At which point you can also consider bumping up replicas of the repo-server.
